src/Uno.UI/UI/Xaml/UIElement.macOS.cs

```diff
@@ -121,125 +103,36 @@ internal Windows.Foundation.Point GetPosition(Point position, global::Windows.UI
 #endif
 		}
 
-		private CGRect ToCGRect(Rect rect)
-		{
-			return new CGRect
-			(
-				(nfloat)(rect.X),
-				(nfloat)(rect.Y),
-				(nfloat)(rect.Width),
-				(nfloat)(rect.Height)
-			);
-		}
-
-		/// <summary>
-		/// Gets the origin point of the <paramref name="view"/> in the clippingParent's 
-		/// coordinate system.
-		/// </summary>
-		/// <param name="view">The view to get the point from</param>
-		/// <param name="parentView">The view for which to get the adjusted coordinates from</param>
-		/// <returns></returns>
-		private static CGPoint ConvertOriginPointToView(NSView view, NSView parentView)
-		{
-			var value = CGPoint.Empty;
-			var current = view;
-
-			do
-			{
-				if (current is FrameworkElement fr)
-				{
-					value.X += (nfloat)fr.AppliedFrame.X;
-					value.Y += (nfloat)fr.AppliedFrame.Y;
-				}
-				else
-				{
-					value.X += current.Frame.X;
-					value.Y += current.Frame.Y;
-				}
-
-				current = current.Superview;
-
-			} while (current != null && current != parentView);
-
-			return value;
-		}
-
-#region DoubleTapped event
-		private void RegisterDoubleTapped(DoubleTappedEventHandler handler)
-		{
-			LogRegisterPointerCanceledNotImplemented();
-		}
-
-		private void UnregisterDoubleTapped(DoubleTappedEventHandler handler)
-		{
-			LogRegisterPointerCanceledNotImplemented();
-		}
-#endregion
-
-#region PointerCanceled event
-		private void RegisterPointerCanceled(PointerEventHandler handler)
-		{
-			LogRegisterPointerCanceledNotImplemented();
-		}
-
-		private void UnregisterPointerCanceled(PointerEventHandler handler)
-		{
-			LogUnregisterPointerCanceledNotImplemented();
-		}
-#endregion
-
-#region PointerExited event
-		private void RegisterPointerExited(PointerEventHandler handler)
-		{
-			LogRegisterPointerExitedNotImplemented();
-		}
+#if DEBUG
+		public string ShowLocalVisualTree(int fromHeight) => AppKit.UIViewExtensions.ShowLocalVisualTree(this, fromHeight);
+#endif
 
-		private void UnregisterPointerExited(PointerEventHandler handler)
-		{
-			LogUnRegisterPointerExitedNotImplemented();
-		}
-#endregion
+		/// <inheritdoc />
+		public override bool AcceptsFirstResponder() 
+			=> true; // This is required to receive the KeyDown / KeyUp. Note: Key events are then bubble in managed.
 
-#region PointerPressed event
-		private void RegisterPointerPressed(PointerEventHandler handler)
+		private protected override void OnNativeKeyDown(NSEvent evt)
 		{
-			LogRegisterPointerPressedNotImplemented();
-		}
+			Console.WriteLine($"[DOWN] Got key: {evt.KeyCode} | char: {evt.Characters} | char_no_modif: {evt.CharactersIgnoringModifiers}");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this logging statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5545012)